_posts/2019-02-28-getting-started-with-webpack.md

```diff
@@ -0,0 +1,141 @@
+
+The intital project state can be found [here](https://github.com/alexoragz/post-webpack-basics/tree/working-with-only-javascript) on github. We will progress through relevant branches as we continue.
+
+___
+Project summary:
+- `src/tracker.js` is a utility that tracks usages for analytics
+- `src/accordion.js` and `src/counter.js` respectively contain the logic for an accordion and a counter. They both import and make use of the tracker utility.
+- `src/index.js` is our entry file and pulls in the accordion and counter files for usage on the webpage.
+- Finally, the `dist` directory contains an `index.html` page which inludes a script link to a `main.js` file (our bundle - which will be generated by webpack). 
+
+___
+
+Downloading the project and running `webpack` after installing the dependencies (via `npm install`) should result in a new `main.js` file being added to the `dist` directory.
+
+What has happened here? 
+
+The `webpack` command executes the webpack CLI in your project folder with some reasonable defaults, some of which are:
+
++ Your entry file is located at `src/index.js` 
++ You want the output in a `dist/main.js` file --> this is your resulting bundle (Which we already included in our `index.html`)
+
+Opening `index.html` in your browser should result in the content and console logs resembling the screen shot below:
+
+![Result](/images/getting-started-with-webpack/counter_accordion.png "Web app")
+
+What if we wanted to take it a step further? Our page is looking pretty bland. Lets add some style sheets to make things look nice. 
+  
+___
+
+Summary of changes: 
++ Reorganized the scripts into separate directories
+  -  `src/accordion.js` is now `src/accordion/accordion.js`
+  -  `src/counter.js` is now `src/counter/counter.js`
+  -  `src/tracker.js` is now `src/utils/tracker.js`
+
++ Added style sheets and imported them into the scripts.
+  -  added a new stylesheet `src/accordion/accordion.css` and imported it into `src/accordion/accordion.js`
+  -  added a new stylesheet `src/counter/counter.css` and imported it into `src/counter/counter.js`
+
+The updates can be found in the [broken-importing-styles-no-loader](https://github.com/alexoragz/post-webpack-basics/tree/broken-importing-styles-no-loader) branch. 
+
+___
+
+Running `webpack` to get a new bundle now should give the following error message:
+
+![Result](/images/getting-started-with-webpack/missing_loader_error.png "Loader error")
+
+This is webpack telling us that it cannot handle the css files and suggesting that we may need a loader to do it for us.
+
+### The fix:
+
+In order to get the styles working, we need to setup loaders for the css. We will rely on `css-loader` (for parsing the css file) and `style-loader` (for injecting the resulting `<style>` tags into our html page).
+
++ First run `npm install --save-dev css-loader style-loader` to add the npm packages
++ then create a `webpack.config.js` file at the root of your project with the following:
+
+```
```

Comment:
This may provide better formatting? I haven't looked at the rendered output.

```suggestion
```javascript
```

Reply:
Yes it does. thanks!